test/test-vectors.ts

```diff
@@ -26,11 +26,12 @@ const artifacts2x2 = {
 } as ProverArtifacts
 
 interface TestVector {
-  proof: Proof;
+  proof?: Proof; // Optional as it must be generated by prove() first
```

Comment:
I think if it's a test vector, it means the proof should be here and fixed from an external source (as I understand it is?). Otherwise it falls under the class of test that check internal consistency between `prove` and `verify`. But perhaps I'm misunderstanding what the test vectors here are :)

Reply:
I see. So you think it should not be allowed to be generated by a `prove()` call, but that all test vectors should be pre-generated by the same source (for ex. the engine test/logs mentioned in the readme).